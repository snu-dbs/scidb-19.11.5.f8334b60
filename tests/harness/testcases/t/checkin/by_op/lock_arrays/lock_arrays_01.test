--setup
--start-query-logging

--test

# Demonstrate that the global array lock is installed and removed
# with lock_arrays(true), lock_arrays(false), respectively.
--shell --store --command "psql -e -d ${DB_NAME} -U ${DB_USER} -p ${DB_PORT} -h ${DB_HOST} -t -c 'select count(*) from array_version_lock'"
lock_arrays(true)
--shell --store --command "psql -e -d ${DB_NAME} -U ${DB_USER} -p ${DB_PORT} -h ${DB_HOST} -t -c 'select count(*) from array_version_lock'"
lock_arrays(false)
--shell --store --command "psql -e -d ${DB_NAME} -U ${DB_USER} -p ${DB_PORT} -h ${DB_HOST} -t -c 'select count(*) from array_version_lock'"

--cleanup
--stop-query-logging
