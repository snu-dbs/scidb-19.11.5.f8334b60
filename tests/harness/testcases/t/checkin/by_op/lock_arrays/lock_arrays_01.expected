SCIDB QUERY : <psql -e -d ${DB_NAME} -U ${DB_USER} -p ${DB_PORT} -h ${DB_HOST} -t -c 'select count(*) from array_version_lock'>
select count(*) from array_version_lock
     0


SCIDB QUERY : <lock_arrays(true)>
Query was executed successfully

SCIDB QUERY : <psql -e -d ${DB_NAME} -U ${DB_USER} -p ${DB_PORT} -h ${DB_HOST} -t -c 'select count(*) from array_version_lock'>
select count(*) from array_version_lock
     1


SCIDB QUERY : <lock_arrays(false)>
Query was executed successfully

SCIDB QUERY : <psql -e -d ${DB_NAME} -U ${DB_USER} -p ${DB_PORT} -h ${DB_HOST} -t -c 'select count(*) from array_version_lock'>
select count(*) from array_version_lock
     0


